src/provider/controller/IntentHandler.ts

```diff
@@ -47,16 +47,35 @@ export class IntentHandler {
     }
 
     private async raiseWithTarget(intent: IntentWithTarget): Promise<IntentResolution> {
-        const appInfo = await this._directory.getAppByName(intent.target);
+        let appInfo: Application|null;
 
-        if (!appInfo) {
-            throw new FDC3Error(ResolveError.TargetAppNotInDirectory, `No app in directory with name: ${intent.target}`);
-        }
+        const appWindow = this._model.findWindowByAppName(intent.target);
+
+        if (appWindow) {
+            // Target app is running -> ensure that intent is registered in model
+            if (!appWindow.hasIntentListener(intent.type)) {
```

Comment:
Feels odd to do this check here, but in the same method to trust that non-running apps will register listeners. I feel like we should leave this to be caught by `ensureReadyToReceiveIntent`. And what if raiseWithTarget is called during the 5 seconds we allow for the intent listener to be added?

Reply:
That's an excellent point. Will delegate that check to `fireIntent` and eventually throw the `Timeout waiting for intent listener to be added` if appropriate.